src/cli.rs

```diff
@@ -0,0 +1,116 @@
+use clap::{Arg, Command};
+
+#[derive(Debug)]
+pub struct DataloggerSettings {
+    pub directory: String,
+    pub filename: String,
+    pub interval: u64,
+}
+
+#[derive(Debug)]
+pub struct MonitorSettings {
+    pub interval: u64,
+}
+
+#[derive(Debug)]
+pub struct ServerSettings {
+    pub port: u16,
+}
+
+pub fn parse_args() -> (DataloggerSettings, MonitorSettings, ServerSettings) {
+    let matches = Command::new("Navigator Assistant")
+        .version("1.0")
+        .author("BlueRobotics")
+        .about("Start your navigator assistant server")
+        .arg(
+            Arg::new("datalogger_directory")
+                .long("datalogger-directory")
+                .required(false),
+        )
+        .arg(
+            Arg::new("datalogger_filename")
+                .long("datalogger-filename")
+                .required(false),
+        )
+        .arg(
+            Arg::new("datalogger_rate")
+                .long("datalogger-rate")
+                .value_parser(clap::value_parser!(f64))
+                .required(false),
+        )
+        .arg(
+            Arg::new("datalogger_enable")
+                .long("datalogger-enable")
+                .value_parser(clap::value_parser!(bool))
+                .required(false),
+        )
+        .arg(
+            Arg::new("monitor_rate")
+                .long("monitor-rate")
+                .value_parser(clap::value_parser!(f64))
+                .required(false),
+        )
+        .arg(
+            Arg::new("monitor_enable")
+                .long("monitor-enable")
+                .value_parser(clap::value_parser!(bool))
+                .required(false),
+        )
+        .arg(
+            Arg::new("server_port")
+                .long("server-port")
+                .value_parser(clap::value_parser!(u16))
+                .required(false),
+        )
+        .get_matches();
+
+    let datalogger_directory = matches
+        .get_one::<String>("datalogger_directory")
+        .map(|d| d.to_string())
+        .unwrap_or("./".to_string());
```

Comment:
Pretty sure we should add the default values when specifying the argument instead of leaving them as optionals and using these fallbacks. The benefit is that it will show the default values for each argument in the help, and it will be easier to maintain.

Reply:
Plan to fix it changing the clap use
1. https://github.com/bluerobotics/navigator-web-assistant/issues/5